### PR TITLE
Intermittent beginning balance creation

### PR DESCRIPTION
**Pull request by ToddE123**
The Account beginning balance is not always getting properly created due to Get() being used instead of FindFirst(). This PR changes Get() to FindFirst().

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al [hunk 1]
```diff
@@ -74,6 +74,65 @@ codeunit 4017 "GP Account Migrator"
...
 
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"GL Acc. Data Migration Facade", 'OnCreateOpeningBalanceTrx', '', true, true)]
+    procedure OnCreateOpeningBalanceTrx(VAR Sender: Codeunit "GL Acc. Data Migration Facade"; RecordIdToMigrate: RecordId)
+    var
+        GPAccount: Record "GP Account";
+    begin
+        if RecordIdToMigrate.TableNo() <> Database::"GP Account" then
+            exit;
+
+        GPAccount.Get(RecordIdToMigrate);
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:85]: Should we have if not GPAccount.Get(RecordIdToMigrate) then 
                               exit;

Just to be on the safe side?

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al [hunk 2]
```diff
@@ -74,6 +74,65 @@ codeunit 4017 "GP Account Migrator"
...
+        GPGL10111.SetRange(PERIODID, 0);
+        GPGL10111.SetRange(YEAR1, InitialYear);
+        if not GPGL10111.FindFirst() then
+            exit;
+
+        BeginningBalance := GPGL10111.PERDBLNC;
+
+        PostingGroupCode := PostingGroupCodeTxt + format(InitialYear) + 'BB';
+        GPFiscalPeriods.SetRange(YEAR1, InitialYear);
+        if GPFiscalPeriods.FindFirst() then begin
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:117]: Minor if not findfrist then
exit.

Then we have less nesting

**Review by nikolakukrika — APPROVED** [review]

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al (hunk 1 above)
Comment (ToddE123) [reply Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:85]: There appears to be an issue with the merge I did from main as there was only to be a single line changed. I believe we covered this same question in a previous PR (https://github.com/microsoft/ALAppExtensions/pull/19202). Perhaps I was mistaken as to when that PR was going into BC. This PR is only needed to fix that PR.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 45998.

**Pull request merged**